### DISCUSSION TITLE
How can i use Fooocus with NVIDIA Tesla T4 and CUDA 10.1 on my linux server?

### DISCUSSION THREAD

**Discussion by MoonHeritage** [lllyasviel/Fooocus#2572] [Q&A] [answered]
I'm trying to run Fooocus on ubuntu linux x64 Microsoft Azure server  (also got same problem with DigitalOcean servers) with **NVIDIA Tesla T4** card on it. I'm having **CUDA 10.1** on it. 
When running Fooocus i've got pytorch error that it is not compatitable with my CUDA version. Than i studied pytorch docs (https://github.com/pytorch/pytorch/blob/main/RELEASE.md) and found that **i need CUDA 11.8**. But i found no info in internet how to upgrade my CUDA for this card to this version (according to NVIDIA website (https://www.nvidia.com/Download/driverResults.aspx/160640/en-us/) last drivers for it was at 2020 year and last version of CUDA for this card is 10.2., according to chatGPT it is 11.0, but it doesn't know how to install it). Downgrading pytorch also do not works.

Seems like it should be common trouble..

So what should i do to fix this version problem? Is it possible to upgrade CUDA for this card and how?

Does anybody got same issues? Do you need any additional info to help me solve this problem? 

Thank you very much!

**Comment by mashb1t** [comment, marked as answer]
In general, T4 supports CUDA 12.1 and higher if i'm not completely mistaken as this is what Fooocus uses in the Colab playbook (see readme). Also see https://docs.nvidia.com/deploy/cuda-compatibility/index.html / https://www.nvidia.com/Download/driverResults.aspx/200625/en-us/
A driver update is the solution here. There are multiple later drivers available when not selecting "any" as cuda toolkit:

12.4: https://www.nvidia.com/Download/driverResults.aspx/222684/en-us/
12.2 (recommended): https://www.nvidia.com/Download/driverResults.aspx/221957/en-us/
12.0: https://www.nvidia.com/Download/driverResults.aspx/214092/en-us/
11.7: https://www.nvidia.com/Download/driverResults.aspx/200625/en-us/

![image](https://github.com/lllyasviel/Fooocus/assets/9307310/2e4529c5-ead8-4b65-bae7-e59e620bb2fb)

Please check out the driver with CUDA 12.2. Happy to hear your feedback if it worked or not.

**Comment by MoonHeritage** [comment]
That really works, thank you very much!
